src/iris/api.py

```diff
@@ -3129,78 +3195,89 @@ def json_error_serializer(req, resp, exception):
+    api.add_route('/v0/applications/{app_name}', Application())
+    api.add_route('/v0/applications', Applications())
 
-    app.add_route('/v0/modes', Modes())
+    api.add_route('/v0/priorities', Priorities())
 
-    app.add_route('/v0/applications/{app_name}/quota', ApplicationQuota())
-    app.add_route('/v0/applications/{app_name}/stats', ApplicationStats())
-    app.add_route('/v0/applications/{app_name}/key', ApplicationKey())
-    app.add_route('/v0/applications/{app_name}/incident_emails', ApplicationEmailIncidents())
-    app.add_route('/v0/applications/{app_name}/rename', ApplicationRename())
-    app.add_route('/v0/applications/{app_name}', Application())
-    app.add_route('/v0/applications', Applications())
+    api.add_route('/v0/response/gmail', ResponseGmail(iris_sender_app))
+    api.add_route('/v0/response/gmail-oneclick', ResponseGmailOneClick(iris_sender_app))
+    api.add_route('/v0/response/twilio/calls', ResponseTwilioCalls(iris_sender_app))
+    api.add_route('/v0/response/twilio/messages', ResponseTwilioMessages(iris_sender_app))
+    api.add_route('/v0/response/slack', ResponseSlack(iris_sender_app))
+    api.add_route('/v0/twilio/deliveryupdate', TwilioDeliveryUpdate())
 
-    app.add_route('/v0/priorities', Priorities())
+    api.add_route('/v0/stats', Stats())
 
-    app.add_route('/v0/response/gmail', ResponseGmail(iris_sender_app))
-    app.add_route('/v0/response/gmail-oneclick', ResponseGmailOneClick(iris_sender_app))
-    app.add_route('/v0/response/twilio/calls', ResponseTwilioCalls(iris_sender_app))
-    app.add_route('/v0/response/twilio/messages', ResponseTwilioMessages(iris_sender_app))
-    app.add_route('/v0/response/slack', ResponseSlack(iris_sender_app))
-    app.add_route('/v0/twilio/deliveryupdate', TwilioDeliveryUpdate())
+    api.add_route('/healthcheck', Healthcheck(healthcheck_path))
+    return api
 
-    app.add_route('/v0/stats', Stats())
 
-    app.add_route('/healthcheck', Healthcheck(healthcheck_path))
+def get_api(config):
+    logging.basicConfig(format='[%(asctime)s] [%(process)d] [%(levelname)s] %(name)s %(message)s',
+                        level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S %z')
+    db.init(config)
+    spawn(update_cache_worker)
+    init_plugins(config.get('plugins', {}))
+    init_validators(config.get('validators', []))
+    healthcheck_path = config['healthcheck_path']
+    iris_sender_app = config['sender'].get('sender_app')
+
+    debug = False
+    if config['server'].get('disable_auth'):
+        debug = True
+    master_sender = config['sender'].get('master_sender', config['sender'])
+    master_sender_addr = (master_sender['host'], master_sender['port'])
+    # all notifications go through master sender for now
+    app = construct_falcon_api(
+        debug, healthcheck_path, iris_sender_app, master_sender_addr)
 
     # Need to call this after all routes have been created
     app = ui.init(config, app)
```

Comment:
To avoid UI routes. That's why I used different names to distinguish them: `app` = `api` + `ui`.